### PR TITLE
feat(es7): Response's isAccurate method

### PR DESCRIPTION
**Pull request by theus77**
|Q              |A  |
|---------------|---|
|Bug fix?       |N|
|New feature?   |Y|	
|BC breaks?     |N|	
|Deprecations?  |N|	
|Fixed tickets  |N|

### CODE REVIEW

File: src/Elasticsearch/Document/Document.php [hunk 1]
```diff
@@ -12,14 +14,25 @@ class Document implements DocumentInterface
...
+    /** @var string */
+    private $index;
+    /** @var array<string, mixed> */
+    private $raw;
+    /** @var string|null*/
+    private $highlight;
 
     /**
-     * @param array{_id: string, _type: ?string, _source: array} $document
+     * @param Result|array{string, mixed} $document
```
Comment (theus77) [inline src/Elasticsearch/Document/Document.php:25]:
```suggestion
     * @param Result|array<string, mixed> $document
```

**Review by theus77 — COMMENTED** [review]
Almost ready

**Review by theus77 — COMMENTED** [review]
Ok to me

File: src/Elasticsearch/Document/Document.php [hunk 2]
```diff
@@ -12,14 +14,25 @@ class Document implements DocumentInterface
...
+    private $raw;
+    /** @var string|null*/
+    private $highlight;
 
     /**
-     * @param array{_id: string, _type: ?string, _source: array} $document
+     * @param Result|array<string, mixed> $document
      */
-    public function __construct(array $document)
+    public function __construct($document)
```
Comment (Davidmattei) [inline src/Elasticsearch/Document/Document.php:27]: private construct? And to static constructs? ::fromArray, ::fromResult ?

File: src/Elasticsearch/Aggregation/ElasticaAggregation.php [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    /**
+     * @param array<string, mixed> $param
+     */
+    public function setConfig(string $basename, array $param): void
+    {
+        $this->basename = $basename;
+        $this->setParams($param);
+    }
+
+    public function _getBaseName(): string
```
Comment (Davidmattei) [inline src/Elasticsearch/Aggregation/ElasticaAggregation.php:26]:
```suggestion
    public function getBaseName(): string
```

**Review by Davidmattei — CHANGES_REQUESTED** [review]

**Review by Davidmattei — APPROVED** [review]

File: src/Elasticsearch/Document/Document.php (hunk 2 above)
Comment (theus77) [reply src/Elasticsearch/Document/Document.php:27]: It was also an idea, I was not sure what was the better approche, but It like your solution

Comment (theus77) [reply src/Elasticsearch/Document/Document.php]: Done

Comment (Davidmattei) [reply src/Elasticsearch/Document/Document.php]: Perfect!

**Pull request merged**